src/care_communication/api/auth/kiosk_auth.py

```
@@ -1,37 +1,43 @@
+            raise AuthenticationFailed("Invalid encounter ID.")
+
+        patient = encounter.patient
+
+        birth_year_valid = False
+        phone_valid = False
+
+        if birth_year:
+            try:
+                birth_year_valid = patient.year_of_birth == int(birth_year)
+            except (TypeError, ValueError):
+                raise AuthenticationFailed("Enter a valid birth year.")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)